packages/office-ui-fabric-react/src/components/HoverCard/HoverCard.tsx

```diff
@@ -68,6 +68,7 @@ export class HoverCard extends BaseComponent<IHoverCardProps, IHoverCardState> {
           this.setState({
             mode: ExpandingCardMode.expanded
           });
+          this.props.onCardExpand && this.props.onCardExpand();
         }, this.props.expandedCardOpenDelay!);
         this.props.onCardVisible && this.props.onCardVisible();
```

Comment:
Thinking maybe you don't need this if you moved that onCardVIsible inside the setState. In fact, maybe that should be inside the `setState` callback

Reply:
no, onCardVisible is different with onCardExpand, hover card has 2 parts, the compact card and expand card. onCardVisible  will be call when compact card is opened, and onCardExpand should be called when expand card is opened.

Reply:
So if you want to log the expand time, then you should at least put the onCardExpand(ed?) to be inside a callback of `setState`

Reply:
👍 to @kenotron's suggestion of invoking within the `setState` callback. This should guarantee that it fires after the component is rendered with that state.

>setState() does not always immediately update the component. It may batch or defer the update until later. This makes reading this.state right after calling setState() a potential pitfall. Instead, use componentDidUpdate or a setState callback (setState(updater, callback)), either of which are guaranteed to fire after the update has been applied. If you need to set the state based on the previous state, read about the updater argument below.
>
>https://reactjs.org/docs/react-component.html#setstate